torchinductor/graph.py

```diff
@@ -151,7 +151,7 @@ def visit(value):
     def add_tensor_constant(self, data):
         def allocate():
             for name, value in self.constants.items():
-                if data is value:
+                if data.size() == value.size() and torch.allclose(data, value):
```

Comment:
```suggestion
                if (data.size() == value.size() and 
                    data.stride() == value.stride() and
                    data.dtype == value.dtype and
                    data.device == value.device and
                    torch.eq(data, value).all()):
```

Reply:
Surprised device matters, I thought we had code like `constant_name()` that handles consts devices, but otherwise this looks like an improvement yea!